ksqldb-engine/src/test/java/io/confluent/ksql/structured/SchemaKSourceFactoryTest.java

```diff
@@ -336,7 +339,27 @@ public void shouldCreateNonWindowedTableSourceV2WithNewPseudoColumnVersionIfNoOl
     );
 
     // Then:
-    assertThat(((TableSource) result.getSourceStep()).getPseudoColumnVersion(), equalTo(NEW_PSEUDOCOLUMN_VERSION));
+    assertThat(((TableSource) result.getSourceStep()).getPseudoColumnVersion(), equalTo(SystemColumns.CURRENT_PSEUDOCOLUMN_VERSION_NUMBER));
+    assertValidSchema(result);
+  }
+
+  //this state shouldn't exist in normal use, testing for completeness
```

Comment:
Clarify what this comment means -- it's only true today because TableSourceV2 was introduced alongside pseudocolumn versioning, and there are only two pseudocolumn versions today. This test may become relevant in the future (else there's no point adding it).